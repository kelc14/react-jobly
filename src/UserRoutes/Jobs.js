import React from "react";
import "./Jobs.css";
import Job from "./Job";

const Jobs = () => {
  return (
    <div>
      <h2 className="Jobs-heading">JOBS</h2>
      <Job />
      <Job />
      <Job />
      <Job />
      <Job />
      <Job />
      <Job />
    </div>
  );
};

export default Jobs;
