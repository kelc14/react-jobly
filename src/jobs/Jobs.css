.Jobs-container {
  width: 70vw;
  background-color: white;
  border: 1px solid black;
  border-radius: 10px;
  margin: 0 auto;
  padding: 20px;
  overflow: hidden;
}

.Jobs-heading {
  margin: 40px auto;
  margin-bottom: 10px;
  font-family: "Ubuntu", sans-serif;
  font-size: 3rem;
  font-weight: bold;
  color: goldenrod;
  text-align: center;
  letter-spacing: 5px;
  text-shadow: 2px 7px 5px rgba(0, 0, 0, 0.3),
    0px -4px 10px rgba(255, 255, 255, 0.3);
}

.Jobs-error {
  color: goldenrod;
  font-family: "Ubuntu", sans-serif;
  font-size: 1.5rem;
}
