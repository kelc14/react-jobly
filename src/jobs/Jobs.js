import React, { useEffect, useState } from "react";
import "./Jobs.css";
import Job from "./Job";
import SearchForm from "../common/SearchForm";
import JoblyApi from "../api/api";

/** Jobs component
 *
 * renders
 * - searchform to filter jobs by title
 * - list of all jobs (Job components)
 */

const Jobs = () => {
  const [jobs, setJobs] = useState([]);
  const [companies, setCompanies] = useState([]);

  useEffect(() => {
    // get company data:
    const getCompanies = async () => {
      let results = await JoblyApi.getAllCompanies();
      setCompanies(() => [...results]);
    };
    getCompanies();

    // get jobs data:
    const getJobs = async () => {
      let results = await JoblyApi.getAllJobs();
      setJobs(() => [...results]);
    };
    getJobs();
  }, []);

  const searchJobs = async (data) => {
    const params = { title: data.search };
    let res = await JoblyApi.getAllJobs(params);
    setJobs(() => [...res]);
  };

  return (
    <div>
      <h2 className="Jobs-heading">JOBS</h2>
      <SearchForm search={searchJobs} />
      {jobs.length === 0 ? (
        <p className="Jobs-error">Sorry, no results found.</p>
      ) : (
        jobs.map((job) => (
          <Job jobData={job} companiesData={companies} key={job.id} />
        ))
      )}
    </div>
  );
};

export default Jobs;
