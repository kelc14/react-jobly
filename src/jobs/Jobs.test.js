import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Jobs from "./Jobs";

/**  SMOKE TEST */
it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <Jobs />
    </MemoryRouter>
  );
});

/**  SNAPSHOT TEST */
it("renders without crashing", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <Jobs />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
