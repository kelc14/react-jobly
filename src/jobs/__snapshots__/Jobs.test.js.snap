// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="Jobs-heading"
    >
      JOBS
    </h2>
    <div
      class="SearchForm"
    >
      <form>
        <input
          class="SearchForm-input"
          id="search"
          name="search"
          placeholder="Enter search term."
          type="text"
          value=""
        />
        <button
          class="SearchForm-btn"
        >
          Search
        </button>
      </form>
    </div>
    <p
      class="Jobs-error"
    >
      Sorry, no results found.
    </p>
  </div>
</DocumentFragment>
`;
